samples/ProjectGrandmaster.Unity/Assets/Script/ChessPiece/MoveDataStructure.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.USYD.ChessPiece
+{
+    public class MoveDataStructure : MonoBehaviour
```

Comment:
This appears to be following a "Singleton" pattern, and if so, what I would recommend is updating it a bit to make it cleaner.

```csharp
public class ASingleton
{
  private static ASingleton instance;
  public static ASingleton Instance 
  {
    get
    {
      if (instance == null)
      {
        instance = new ASingleton();
      }

      return instance;
    }
  }

   private List<bool> eliminated = new List<bool>();
   private List<GameObject> eliminatedObjects = new List<GameObject>();
   ...

  // Private constructor to avoid public instantiation
  private ASingleton() { }

}
```

Reply:
Updated.